src/contexts/SettingsProvider.tsx

```diff
@@ -146,29 +147,28 @@ export function SettingsContextProvider({ children }: { children: any }) {
   );
 
   async function toggleCollectibleVisibility(nft: NftTokenWithBalance) {
-    const key = nft.address;
-    const collectiblesVisibility = settings?.collectiblesVisibility ?? {};
+    const key = `${nft.address}-${nft.tokenId}`;
+    const visibility = settings?.collectiblesVisibility ?? {};
+    // We used to (wrongly) index by address only.
+    const isHidden = (visibility[key] ?? visibility[nft.address]) === false;
```

Comment:
Having a schema upgrade would be the best for this, but I don't think it's worth the effort (+ we don't have the `tokenId` in the current storage data anyway). After toggling once it "fixes itself" for all NFTs sharing a given `address`.